commit 3181dd8f55dc7fbeeb45467a4eef2a84de3a5710
Author: Michael Still <mikal@stillhq.com>
Date:   Sat Jan 10 06:45:16 2026 +1100

    Re-enable SPICE support on Debian.
    
    The move to Debian Trixie for Kolla master accidentally broke SPICE
    consoles for Nova (both Horizon and spice-direct). This is because Debian
    and Ubuntu now package the SPICE support in a separate package. Without
    this package, an attempt to start an instance in Nova requiring SPICE will
    result in messages like this in the nova-libvirt logs:

        libvirt.libvirtError: unsupported configuration: spice graphics are
        not supported with this QEMU

    And the instance will enter an error state.

    The fix is to install the `qemu-system-modules-spice` package as well in
    the nova-libvirt image. I do not believe this affects Debian Bookworm.
    
    Closes-Bug: #2138186
    Depends-On: https://review.opendev.org/c/openstack/kolla/+/972437
    Change-Id: Ib4ced4a04353b9c11c6f0abb7cfa91001b239e53
    Signed-off-by: Michael Still <mikal@stillhq.com>

diff --git a/docker/nova/nova-libvirt/Dockerfile.j2 b/docker/nova/nova-libvirt/Dockerfile.j2
index d65d31584..88e4a52d6 100644
--- a/docker/nova/nova-libvirt/Dockerfile.j2
+++ b/docker/nova/nova-libvirt/Dockerfile.j2
@@ -54,6 +54,7 @@ LABEL maintainer="{{ maintainer }}" name="{{ image_name }}" build-date="{{ build
         'python3-rbd',
         'qemu-block-extra',
         'qemu-system',
+        'qemu-system-modules-spice',
         'qemu-utils',
         'sasl2-bin',
         'swtpm',
diff --git a/releasenotes/notes/debian-trixie-spice-814929ed1efbcad1.yaml b/releasenotes/notes/debian-trixie-spice-814929ed1efbcad1.yaml
new file mode 100644
index 000000000..3874c534a
--- /dev/null
+++ b/releasenotes/notes/debian-trixie-spice-814929ed1efbcad1.yaml
@@ -0,0 +1,8 @@
+---
+fixes:
+  - |
+    Debian Trixie container images require an additional operating
+    system package for SPICE consoles to function correctly. Without
+    this package instances requesting a SPICE console will fail to
+    start with an error message. Further details are available in
+    the bug report at `LP#2138186 <https://launchpad.net/bugs/2138186>`__.
